integration_tests/tests/s3-invoice-end-to-end-tests.ts

```diff
@@ -0,0 +1,122 @@
+  checkIfS3ObjectExists,
+  deleteS3Object, deleteS3Objects,
+  listS3Objects,
+} from "../../src/handlers/int-test-support/helpers/s3Helper";
+import { waitForTrue } from "../../src/handlers/int-test-support/helpers/commonHelpers";
+import { randomInvoice, randomLineItems } from "../../src/handlers/int-test-support/helpers/mock-data/invoice";
+import { createInvoiceInS3 } from "../../src/handlers/int-test-support/helpers/mock-data/invoice/helpers";
+import {queryObject, startQueryExecutionCommand} from "../../src/handlers/int-test-support/helpers/athenaHelper";
+
+const prefix = resourcePrefix();
+const testStartTime = new Date();
+
+describe("\n Happy path - Upload valid mock invoice pdf and verify data is seen in the billing view\n", () => {
+  const storageBucket = `${prefix}-storage`;
+  const standardisedFolderPrefix = "btm_billing_standardised";
+  const databaseName = `${prefix}-calculations`;
+
+  beforeAll(async () => {
+    await deleteS3Objects({
+      bucketName: storageBucket,
+      prefix: standardisedFolderPrefix,
+    });
+  });
+
+  test("raw-invoice-textract-data and storage buckets should contain textracted and standardised data file for uploaded valid pdf file in raw-invoice-pdf bucket and should move the original raw invoice to successful folder in s3 raw-invoice-pdf bucket", async () => {
+    const passportCheckItems = randomLineItems(8, { description: "passport check"});
+    const addressCheckItems = randomLineItems(10, { description: "address check"});
+    const invoice = randomInvoice({ lineItems: [ ...passportCheckItems, ...addressCheckItems ]});
+    const expectedSubtotals = [ invoice.getSubtotal("address check"), invoice.getSubtotal("passport check") ];
+    const expectedQuantities = [ invoice.getQuantity("address check"), invoice.getQuantity("passport check") ];
+    const expectedServices = [ "Address check", "Passport check" ];
+    const filename = `raw-Invoice-${Math.random()
+      .toString(36)
+      .substring(2, 7)}-validFile`;
+    const s3Object = await createInvoiceInS3({
+      invoiceData: invoice,
+      filename: `${filename}.pdf`});
+    const checkRawPdfFileExists = await checkIfS3ObjectExists(s3Object);
+    expect(checkRawPdfFileExists).toBeTruthy();
+
+
+    // Wait for the invoice data to have been written, to some file in the standardised folder.
+    const haveInvoiceLineItemsBeenGenerated = async (): Promise<boolean> => {
+      const result = await listS3Objects({
+        bucketName: storageBucket,
+        prefix: standardisedFolderPrefix,
+      });
+      if (result.Contents === undefined) {
+        return false;
+      }
+      return result.Contents.some((s3Object) =>
+        s3Object.LastModified !== undefined &&
+        new Date(s3Object.LastModified) >= testStartTime
+      );
+    };
+    const invoiceLineItemsGenerated = await waitForTrue(
+      haveInvoiceLineItemsBeenGenerated,
+      1000,
+      21000
+    );
+    expect(invoiceLineItemsGenerated).toBeTruthy();
```

Comment:
```suggestion
    await poll(
      async () =>
        await listS3Objects({
          bucketName: storageBucket,
          prefix: standardisedFolderPrefix,
        }),
      ({ Contents }) =>
        !!Contents?.some(
          (s3Object) =>
            s3Object.LastModified !== undefined &&
            new Date(s3Object.LastModified) >= testStartTime
        ),
      {
        timeout: 21000,
        nonCompleteErrorMessage:
          "Invoice data never appeared in standardised folder",
      }
    );
```

Using this gives you better output if the test fails. With `waitForTrue` you just get 
```
Expected: true
Received: false
```